.github/workflows/Release.yml

```diff
@@ -0,0 +1,57 @@
+name: Release CI
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - "Project.toml"
+jobs:
+  branch:
+    name: "Create Release Branch"
+    runs-on: ubuntu-20.04-16core
+    strategy:
+      fail-fast: false
+    steps:
+        - uses: actions/checkout@v3
+        - uses: julia-actions/setup-julia@v1
+          with:
+            version: "1.8"
+            arch: "x64"
+        - uses: julia-actions/cache@v1
+        - name: Check Project version
+          id: version_check
+          shell: julia --color=yes --project {0}
+          run: |
+            using Pkg.Types
+            project = read_project("Project.toml")
+            latest_tag = parse(VersionNumber, readchomp(`git describe --tags --abbrev=0`))
+            @assert project.version > latest_tag "$(project.version) !> $latest_tag"
+            open(ENV["GITHUB_OUTPUT"], "a") do io
+              println(io, "version=$(project.version)")
+            end
+          continue-on-error: true
+        - name: Checkout new branch
+          if: steps.version_check.outcome = 'success'
+          run: |
+            release_branch="release-v${{ steps.version_check.outputs.version }}"
+            # Error if branch already exists
+            if [[ ! -z $(git ls-remote --heads origin $release_branch) ]]; then
+              echo "Release branch already exists on remote: $release_branch" >&2
+              exit 1
+            fi
+            git checkout -b $release_branch
```

Comment:
Shouldn't this step just create and push the branch to the remote and leave the checking out for another step?

Reply:
Can you elaborate? what's the difference if we checkout then push?